### ISSUE TITLE
default organization repository sort order

### ISSUE THREAD

**Issue by agaida** [go-gitea/gitea#8795]
- Gitea version (or commit ref): current
- all the other things are not relevant for this feature request

## Description
Hi, there was several good bug reports about repository sort order but unfortuately all fall short.
Default sorting order should be configurable - best example is a 'morgue' or 'basement' organization - nobody will care when the last commit was done on dead projects - the natural order here would be maybe alphabetic or rev. alpha. So it would be nice if one would be able to set this per org.

**Comment by ddetton** [comment]
Default sort order should be implemented for everything.  Most important would be orgs and repos.  I don't care what the default is, as long as there is a way to override it.

**Comment by CrysK** [comment]
I'm also looking for a way to default sort, especially for the issue tracker (recently updated first).
Is there a way to set a default meanwhile?

**Comment by micka190** [comment]
Also interested in sorting issues by recently updated first. It's a drag to have to go through multiple issues (sometimes multiple pages) to find the ones that are actively being worked on/discussed

**Comment by bobmagicii** [comment]
yeah i really wish it would default to sorting by alpha on all our orgs, we aren't treating dev as a race and people just need to be able to find things they need in a consistent way that the alphabet provides. such that, my team uses a tampermonkey script to redirect to ?sort=alphabetically if we visit without a sort set. 

in my mind, i feel like the front end should remember that in local storage or something. then like the tampermonkey, its per user choice where they left it.

**Comment by TimoNotonica** [comment]
Same for me. Define default sort order in app.ini and/or ui.

**Comment by levicki** [comment]
I am curious, is there any valid reason why configurable sort order hasn't been implemented for 5 years? I get that it's a cosmetic issue compared to making sure other more important features work but still... it's unfathomable that it hasn't been done yet when it's such a basic feature of any list in any GUI.

**Comment by lunny** [comment]
Should this be a per org level configuration or a system level configuration?

**Comment by micka190** [comment]
Personally, I'd be happy with a feature that was at least repo level or user level. I don't think system-level makes sense since you probably want *some* flexibility (if there's different sorting modes). I'd prefer repo level over org-level, though I'd settle for org-level if it's the only choice.

**Comment by levicki** [comment]
@lunny

Here's what I would like to see:

1. Default to sorting repos and orgs alphabetically everywhere.

Rationale:
Current order by last modified time is bad because people use muscle memory to navigate but repos and orgs keep jumping around as they are updated making consistent navigation impossible. Furthermore, items not sorted alphabetically make it harder to navigate by scrolling or skipping pages which is often faster than using search.

2. Allow setting a system-wide sort default in `app.ini`

Rationale:
Perhaps someone will disagree with my first proposal and they will want to revert to current sort order. Allowing setting a system-wide default in config would let them accomplish that.

3. Store manually selected sort order in user settings

Rationale:
This will allow individual users to override system-wide settings.

Hope that helps.

**Comment by lunny** [comment]
For the first requirement, now we have #30784.

**Comment by levicki** [comment]
> For the first requirement, now we have #30784.

That seems just for one spot in the UI, right?

**Comment by bryanpedini** [comment]
> I don't care what the default is, as long as there is a way to override it.

This. I mean, #2647 changed some defaults, but still no option neither in the database nor in `app.ini` to edit those, only current customizable thing I could find is `ui.EXPLORE_PAGING_DEFAULT_SORT`. No org default, no user default, nothing.

> I do commits on old projects to tweak readmes, but I don't want them surfacing to the top

Also this too, it would be nice if it was possible to define a way for old repos to *not* be considered in the default "recently updated" sorting.
For example, if a repo is archived or some other flag, it should not be considered "recently updated", since it's "old anyways" or somethig.

**Comment by ezzra** [comment]
I am no go developer, but is there any way to sponsor this adaption (only simple default order config) in any way?

**Comment by lunny** [comment]
Maybe related #31921

**Comment by bryanpedini** [comment]
> Maybe related #31921

related commits indeed, however I don't see a call to the app.ini settings for configurable default sort order (am I missing something?)
not an experienced Go programmer but I would really like (when #31921 gets merged) to pick up those changes, dive into the code, and make them configurable via app.ini (maybe?)

**Comment by levicki** [comment]
@lunny 

I can't believe that only explore sort was added after this much time and even that sort is broken &mdash; it sorts upper and lower case separately as if that's how humans ever sort names...

**Comment by bcannoff** [comment]
I just had this request from some of my developers - they would definitely prefer alphabetical repositories as the default.
